### ISSUE TITLE
Expose logp for Bayesian models in brainiak

### ISSUE THREAD

**Issue by mshvartsman** [IntelPNI/brainiak#69]
SRM and (H)TFA, and Bayesian RSA are all Bayesian models. It would be great if they exposed a method that produces a log-likelihood given some data and parameters. This would make easier a number of things: 

- Bayesian model comparison. 
- Decomposing the model structure from the estimation machinery for better reuse of either/both.

**Comment by TuKo** [comment]
@mshvartsman , how would a log-likelihood function would help with model comparison here?
Can you give an expected usage example?

**Comment by mihaic** [comment]
@mshvartsman, can you please reply to Javier?

**Comment by mshvartsman** [comment]
For Bayesian model comparison I guess we would need marginal likelihood, not likelihood given data and parameters, so maybe this is not as useful on this front. 

For the latter situation I'm imagining something like the following: if I estimate a joint hyperprior over some latent cognitive features of behavioral data and latent factors I could alternate doing MAP estimation on the brain factor model, cognitive features, and the hyper-prior. I would be reusing the factor model likelihood but the rest of the structure would be different. 

I'm hoping to put together a simpler version (using PCA instead of a fancy factor model) as a proof of concept, so if you prefer we can close this and wait on it until I have a concrete thing I want to plug one of these factor models into to reopen.

**Comment by lcnature** [comment]
This is a good idea as I also have plan to offer a model comparison capacity in Bayesian RSA. The simple way is to give BIC or AIC score. Doing cross-validation can be better. But fit() seems to be reserved for giving back labels for testing data instead of giving log-likelihood. What function name would be good?
But anyway I prefer to keep this as lower priority for my code as I have a few other things to add first.
